# Cleaning and preprocessing

According to the codebook on [Kaggle](https://www.kaggle.com/datasets/thedevastator/mental-health-in-drug-users-during-covid-19), each respondent's identifier is given in the `hash` column of each data file.  First we load the translated data we saved in the ProblemStatement notebook.

In [1]:
# Import pandas to load the data
import pandas as pd

# Load the first data set
data1 = pd.read_csv("data/translated_data1.csv")

# Load the second data set
data2 = pd.read_csv("data/translated_data2.csv")

Now we can check how many respondents completed both surveys by looking for `hash` entries that appear in both data files.  

In [2]:
# Create a list of the hash ids that appear in both data files
common_hash_ids = []

# Append to the list using a loop
for hash_id in data1['hash'].values:
    if (hash_id in data2['hash'].values)==True:
        common_hash_ids.append(hash_id)

# Count the number of entries in the list
len(common_hash_ids)

7090

Based on the output, there are 7090 respondents who completed both surveys.  We shall create a data frame merging the data from both data files for each of those respondents.

In [3]:
# Create the new data frame
master_data = pd.merge(data1,
                       data2, 
                       how='inner',
                       on="hash")

# Show the new data frame
master_data    

Unnamed: 0_x  0_1  0_2  0_3  0_4  0_5  0_6  0_7  0_8  0_9  ...  \
0                6    4    5    5    1    5    3    5    1    5  ...   
1                7    4    3    5    3    4    3    4    4    4  ...   
2                8    5    5    5    3    3    5    5    1    1  ...   
3                9    4    4    5    4    4    3    5    3    3  ...   
4               10    5    3    2    2    5    5    5    4    4  ...   
...            ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   
7085        157068    5    3    5    2    5    1    5    5    3  ...   
7086        157070    4    5    5    5    2    4    4    5    1  ...   
7087        157071    4    2    3    3    3    4    5    5    3  ...   
7088        157072    2    3    3    3    3    4    5    4    4  ...   
7089        157075    4    2    3    3    4    3    4    5    4  ...   

      12_usually_find_things_to_laugh about  \
0                                         6   
1                                         6   
2                                         6   
3                                         4   
4                                         7   
...                                     ...   
7085                                      7   
7086                                      5   
7087                                      7   
7088                                      4   
7089                                      7   

      12_confidence_in_my_going_through_difficult_times  \
0                                                     6   
1                                                     7   
2                                                     3   
3                                                     2   
4                                                     7   
...                                                 ...   
7085                                                  7   
7086                                                  2   
7087                                                  4   
7088                                                  5   
7089                                                  3   

      12_in_emergencies_you_can_trust_me  \
0                                      7   
1                                      7   
2                                      7   
3                                      6   
4                                      7   
...                                  ...   
7085                                   7   
7086                                   7   
7087                                   7   
7088                                   7   
7089                                   7   

      12_see_situation_several_points_of_view  12_my_life_has_sense  \
0                                           7                     7   
1                                           5                     5   
2                                           6                     6   
3                                           6                     3   
4                                           6                     7   
...                                       ...                   ...   
7085                                        5                     6   
7086                                        6                     1   
7087                                        7                     3   
7088                                        7                     6   
7089                                        5                     4   

      12_I_don't_insist_in_things_that_I_can't_do_anything  \
0                                                     7      
1                                                     6      
2                                                     2      
3                                                     3      
4                                                     5      
...                                                 ...      
7085                                               

As expected, `master_data` has 7090 rows.

Now it would be helpful if the numbered columns from `data1` were replaced with their descriptions.  To do this we can scrape the table of the column descriptions from [the Kaggle page](https://www.kaggle.com/datasets/thedevastator/mental-health-in-drug-users-during-covid-19).

In [5]:
# Import BeautifulSoup to deal with html
import bs4
#from bs4 import BeautifulSoup

# Import requests to get the html
import requests

# This is the url for the html we want
kaggle_url = "https://www.kaggle.com/datasets/thedevastator/mental-health-in-drug-users-during-covid-19"

# Request the html code and assign it to a variable
kaggle_html = requests.get(kaggle_url)

# Make a BeautifulSoup object
kaggle_soup = BeautifulSoup(kaggle_html.content)

NameError: name 'BeautifulSoup' is not defined